### PR TITLE
OpenTelemetry

### PR DESCRIPTION
**Pull request by errordeveloper**

### CODE REVIEW

File: cmd/podinfo/main.go [hunk 1]
```diff
@@ -53,6 +53,7 @@ func main() {
 	fs.Int("stress-cpu", 0, "number of CPU cores with 100 load")
 	fs.Int("stress-memory", 0, "MB of data to load into memory")
 	fs.String("cache-server", "", "Redis address in the format <host>:<port>")
+	fs.String("otel-service-name", "podinfo", "Redis address in the format <host>:<port>")
```
Comment (stefanprodan) [inline cmd/podinfo/main.go:56]: The description of this flag needs to be updated.

File: pkg/api/http.go [hunk 2]
```diff
@@ -33,27 +35,6 @@ func versionMiddleware(next http.Handler) http.Handler {
 	})
 }
 
-// TODO: use Istio tracing package
-// https://github.com/istio/istio/blob/master/pkg/tracing/config.go
-func copyTracingHeaders(from *http.Request, to *http.Request) {
```
Comment (stefanprodan) [inline pkg/api/http.go:38]: Can you please add a unit test for `api/echo` and prove that the OTEL client does forward all these headers?

**Review by stefanprodan — COMMENTED** [review]
Hi @errordeveloper thanks for this PR. I've added some comments please check then out. I see that unit tests are failing in CI with: 

```
--- FAIL: TestChunkedHandler (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xa4d9ed]
```

Can you please fix the tests.

**Comment by rajatvig** [comment]
@errordeveloper Is this port still active? Really interested in this

**Comment by errordeveloper** [comment]
> @errordeveloper Is this port still active? Really interested in this

I made this work for a demo, it just needs a little bit more work before merging, see @stefanprodan's comment... Would you be interested to help out?

**Comment by stefanprodan** [comment]
Besides adding tests for the headers we also need to look into why chunked transfer is broken:

```
--- FAIL: TestChunkedHandler (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xa4d9ed]

goroutine 6 [running]:
testing.tRunner.func1.2(0xb00b40, 0x199f670)
	/opt/hostedtoolcache/go/1.16.10/x64/src/testing/testing.go:1143 +0x332
testing.tRunner.func1(0xc000106d80)
	/opt/hostedtoolcache/go/1.16.10/x64/src/testing/testing.go:1146 +0x4b6
panic(0xb00b40, 0x199f670)
	/opt/hostedtoolcache/go/1.16.10/x64/src/runtime/panic.go:965 +0x1b9
github.com/stefanprodan/podinfo/pkg/api.(*Server).chunkedHandler(0xc0003d74f0, 0xcc6008, 0xc000032180, 0xc00017cf00)
	/home/runner/work/podinfo/podinfo/pkg/api/chunked.go:21 +0x8d
net/http.HandlerFunc.ServeHTTP(0xc0003cf760, 0xcc6008, 0xc000032180, 0xc00017cf00)
	/opt/hostedtoolcache/go/1.16.10/x64/src/net/http/server.go:2049 +0x44
github.com/gorilla/mux.(*Router).ServeHTTP(0xc000180480, 0xcc6008, 0xc000032180, 0xc00017c900)
	/home/runner/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0xd3
github.com/stefanprodan/podinfo/pkg/api.TestChunkedHandler(0xc000106d80)
	/home/runner/work/podinfo/podinfo/pkg/api/chunked_test.go:20 +0x216
testing.tRunner(0xc000106d80, 0xc1a8d0)
	/opt/hostedtoolcache/go/1.16.10/x64/src/testing/testing.go:1193 +0xef
created by testing.(*T).Run
	/opt/hostedtoolcache/go/1.16.10/x64/src/testing/testing.go:1238 +0x2b3
```

**Comment by rajatvig** [comment]
@stefanprodan @errordeveloper Have fixed the failing test and reverted the change to skip copying headers as OpenTelemetry does not support context B3 propagation by default. https://github.com/stefanprodan/podinfo/pull/170 is the PR for reference.

**Comment by stefanprodan** [comment]
Superseded by https://github.com/stefanprodan/podinfo/pull/170

**Pull request closed without merging**